### PR TITLE
Resolve mixins in JSONSchema generator

### LINKED ISSUE #223: JSONSchema generator does not expand mixins correctly
Given following example API Blueprint
```
# Resource [/example]

## Action [GET]

+ Response 200 (application/json)

    + Attributes (Notification Token Full)

# Data Structures

## Notification Token Create (object)
- device_type: `ios`    (string) - The platform. Should be `ios` or `gcm`.
- token: `a-push-token` (string) - The token given by the platform.
- links                 (object, optional)
  - user: 123           (string)

## Notification Token Full (object)
- id: `122` (string)
- Include Notification Token Create
```
the JSON Schema produced is not correct.

JSON:
```
{
 "id":"122",
 "device_type": "ios",
 "token": "a-push-token",
 "links": {
   "user": "123"
 }
}
```

JSON Schema
```
{ 
"$schema": "http://json-schema.org/draft-04/schema#",   
"type": "object",   
"properties": {
    "id": {"type": "string"},
    "type": "object"
    }
}
```

Expected JSON Schema should look more like the following
```
{ 
"$schema": "http://json-schema.org/draft-04/schema#",   
"type": "object",   
"properties": {
    "id": {"type": "string"},
    "device_type": {"type": "string"},
    "token": {"type": "string"},
    "links": {"type": "object"},
    }
}
```

### PR DESCRIPTION
**Pull request by w-vi**
Fixes #223

### CODE REVIEW

File: src/refract/JSONSchemaVisitor.cc [hunk 1]
```diff
@@ -33,6 +33,45 @@ namespace refract
...
+                IElement::MemberElementCollection::const_iterator found = (*it)->attributes.find("resolved");
+
+                if (found == (*it)->attributes.end()) {
+                    continue;
+                }
+
+                const T* resolved = TypeQueryVisitor::as<T>((*found)->value.second);
+
+                if (!resolved) {
+                    throw refract::LogicError("Mixin must refer to same type as parent");
```
Comment (klokane) [inline src/refract/JSONSchemaVisitor.cc]: this should be probable catched one level higher and rethrow it with sourcemap info

**Comment by klokane** [comment]
Patch it self looks OK. We need resolve error info propagation to upper level of application, because in generators (JSON/JSONSchema) we have no required info. Probably we need to create new task for it, because there is same problem in JSON

File: src/refract/JSONSchemaVisitor.cc (hunk 1 above)
Comment (w-vi) [inline src/refract/JSONSchemaVisitor.cc]: That's the problem we have discussed on Friday isn't it? We should take some more general approach to this. I think even in this state it should be merged and the throw should be fixed later in separate PR where we fix other related parts too.

**Comment by w-vi** [comment]
@klokane I agree we should fix the exceptions with sourcemaps in separate PR fixing it in other places too. If it is otherwise good, them merge it please. We should release drafter soon, as some of the changes/fixes are high priority.

**Comment by klokane** [comment]
Ok going to merge.

**Comment by pksunkara** [comment]
:+1:

**Comment by honzajavorek** [comment]
This PR doesn't fully solve the issue. Please see https://github.com/apiaryio/drafter/issues/248.

**Pull request merged**